src/interceptors/exception.interceptor.ts

```diff
@@ -0,0 +1,35 @@
+import {
+  ArgumentsHost,
+  Catch,
+  ExceptionFilter,
+  HttpException,
+  Logger,
+} from '@nestjs/common';
+import { Response } from 'express';
+import { APIResponse } from 'src/utils/apiResponse.utils';
+
+@Catch(HttpException)
+export class HttpExceptionFilter implements ExceptionFilter {
+  private logger = new Logger();
+
+  catch(exception: HttpException, host: ArgumentsHost) {
+    let err = exception['response'];
+    err = err.response ? err.response : JSON.stringify(err);
+    const context = host.switchToHttp();
+    const response = context.getResponse<Response>();
+    const status = exception.getStatus();
+    this.logger.error(
+      `Message : ${err}`,
+      `${context.getRequest().method} ${context.getRequest().url}`,
+    );
+    const error =
+      typeof exception.getResponse() === 'string'
+        ? exception.getResponse()
+        : exception.getResponse() &&
+            typeof exception.getResponse()['message'] === 'object'
+          ? exception.getResponse()['message'].join(', ')
+          : exception.getResponse()['message'];
```

Comment:
PullProofAI

The logic for extracting error messages can be simplified and improved for readability. Consider refactoring this part of the code.
```diff
- let err = exception['response'];
- err = err.response ? err.response : JSON.stringify(err);
+ const errorMessage = exception.getResponse()['message'] || 'Internal Server Error';
- const error =
-   typeof exception.getResponse() === 'string'
-     ? exception.getResponse()
-     : exception.getResponse() &&
-         typeof exception.getResponse()['message'] === 'object'
-       ? exception.getResponse()['message'].join(', ')
-       : exception.getResponse()['message'];
```

33:
Ensure proper error handling by wrapping the response status and JSON serialization in a try-catch block.
```diff
try {
    response.status(status).json(APIResponse.create(0, status, error));
} catch (error) {
    this.logger.error('Error sending response', error);
}
```

### Example response



<!-- This is an auto-generated comment by PullProofAI -->